caddy.conf.d/csp.conf

```diff
@@ -0,0 +1,4 @@
+header / {
+    # CSP (see http://content-security-policy.com/ )
+    Content-Security-Policy "upgrade-insecure-requests; default-src 'none'; script-src 'self'; style-src 'self' 'unsafe-inline'; form-action 'self'; font-src 'self'; frame-ancestors 'self'; base-uri 'self'; connect-src 'self' https://overpass-api.de; img-src 'self' data: https://*.tile.openstreetmap.org; frame-src https://www.youtube-nocookie.com https://player.vimeo.com https://www.dailymotion.com https://www.deezer.com https://www.mixcloud.com https://w.soundcloud.com https://embed.spotify.com"
+}
```

Comment:
Why not add the other headers ?

Strict-Transport-Security, X-XSS-Protection, X-Content-Type-Options, X-Frame-Options, Feature-Policy, Pragma, Referrer-Policy, X-Robots-Tag, -Server, Cache-Control, Access-Control-Allow-Methods, Access-Control-Allow-Origin

Reply:
It's already here: https://github.com/searx/searx-docker/pull/26/files#diff-eb672b1ddd3aa7dae5f2577f38daa271R51.

The csp.conf config file is only useful for the user to change the CSP headers if he disables the image proxy.

Reply:
I've seen but why the only HTTP header in caddy.conf.d is Content-Security-Policy ?

Either:
* put Content-Security-Policy is docker-compose.yaml
* put the other headers in a caddy.conf.d/headers.conf file

May be I miss the point of a different file for the CSP header.

Reply:
If the CSP is put in the `docker-compose.yaml` then when the `docker-compose.yaml` is going to be updated with new changes and the user changed the CSP because he disabled the image proxy there will be a git conflict like you described it (https://github.com/searx/searx-docker#custom-docker-composeyaml):
> Do not modify docker-compose.yaml otherwise you won't be able to update easily from the git repository.

I don't mind putting the other headers into a separate file but is it really useful for the user?
I don't think anyone will want to change the other headers by hand because all of them are "standard" and doesn't need to be customized. This also allows us to be sure that when creating an issue the user will have the same headers as the ones in the repository.
Moreover, if one day we need to generate "dynamically" a header based on the `.env` then we wouldn't be required to remove it from the "headers.conf" file then add it back as a Docker label.